### ISSUE TITLE
Displaying a Message When No Suggestions Provided to AutoSuggest

### ISSUE THREAD

**Issue by BigPrimeNumbers** [moroshko/react-autosuggest#157]
Is it possible to have Autosuggest display a suggestion such as "Nothing found." if an empty array is passed to the Autosuggest component? 

I have it working right now by having `getSuggestions` place a single React element of `<p>Nothing found.</p>` into the suggestions array if none are found when filtering, but this feels hacky, and causes this default suggestion to display for a split second when suggestions are actually found.

**Comment by moroshko** [comment]
I haven't tried this, but what if instead of passing an empty array, you use `multiSection={true}` with:
```js
suggestions={{ title: 'Nothing found', suggestions: [] }}
```
?

**Comment by BigPrimeNumbers** [comment]
Unfortunately this doesn't work, as the empty suggestions array in the object causes `renderSectionTitle` to never be called. I can put an empty object in the suggestions array (within the object) which causes the title to be rendered, but an empty, selectable item underneath the title also appears, which is less than ideal.

![untitled](https://cloud.githubusercontent.com/assets/3743463/14966266/56682d60-1066-11e6-9d87-8f43824a9f4d.png)


Also, I'm assuming you meant to write the above with an array, not an object with an object?

```javascript
result = [{ title: 'No results found', suggestions: [] }]
```

Any other ideas? Thanks for the help!

**Comment by moroshko** [comment]
Have a look at this [Codepen](http://codepen.io/moroshko/pen/wMRmmM).

**Comment by BigPrimeNumbers** [comment]
Thanks for the example, though we were trying to get this to show up within the autosuggest dropdown. I ended up accomplishing this with the single React element of <p>Nothing found.</p> being placed into the suggestions array, and debouncing to prevent it from showing up momentarily before our server could return possible suggestions that could be filtered. Thanks for your time and help with this one!

**Comment by mohsinrasool** [comment]
@BigPrimeNumbers Can you please share working example/codepen for your idea?

**Comment by BigPrimeNumbers** [comment]
The project got a little too sprawling and my final solution too involved for me to easily post it in an example. I can give you the basic gist of it though:

1. Right before I'm about to fire off the network request to search for possibilities, I set a flag that tells the autosuggest component that it is waiting for a response.
2. Fire the network request.
3. While I wait, I render a suggestion that conveys that a search is occurring, e.g. "Searching DB"
4. Once the result returns, I set that flag to false and examine what I got back from the network request, displaying accordingly (e.g. "No Results Found" or the list of results).

I didn't end up using any debounced methods for this issue in particular, instead relying on a flag to know when I should display suggestions (or lack thereof).
